src/Network/Network/VirtualNetwork/Subnet/GetAzureVirtualNetworkSubnetConfigCommand.cs

```diff
@@ -17,44 +17,68 @@
 using System.Linq;
 using System.Management.Automation;
 using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+using Microsoft.Azure.Commands.Network.Common;
 
 namespace Microsoft.Azure.Commands.Network
 {
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkSubnetConfig"), OutputType(typeof(PSSubnet))]
-    public class GetAzureVirtualNetworkSubnetConfigCommand : NetworkBaseCmdlet
-    {
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkSubnetConfig",DefaultParameterSetName = "GetByVirtualNetwork"), OutputType(typeof(PSSubnet))]
+    public class GetAzureVirtualNetworkSubnetConfigCommand : VirtualNetworkBaseCmdlet
+  {
         [Parameter(
             Mandatory = false,
+            ParameterSetName = "GetByVirtualNetwork",
             HelpMessage = "The name of the subnet")]
         public string Name { get; set; }
 
         [Parameter(
             Mandatory = true,
             ValueFromPipeline = true,
+            ParameterSetName = "GetByVirtualNetwork",
             HelpMessage = "The virtualNetwork")]
         public PSVirtualNetwork VirtualNetwork { get; set; }
 
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "GetByResourceId",
+            HelpMessage = "Resource Id to the subnet"
```

Comment:
@lwajswaj would you mind adding `ValueFromPipelineByPropertyName = true` to this parameter to enable piping from `Get-AzResource`?

Reply:
Done